.gitlab/ruby-build-and-test-extra.yml

```diff
@@ -12,7 +12,22 @@
 # We keep ${PROJECT_<MACHINE>_VARIANTS} and ${PROJECT_<MACHINE>_DEPS} So that
 # the comparison with the original job is easier.
 
-# No overridden jobs so far.
+clang_14_0_6:
+  variables:
+    SPEC: " ~shared +openmp +omptask +tests %clang@14.0.6"
+  extends: .build_and_test_on_ruby
+
+gcc_10_3_1:
+  variables:
+    SPEC: " ~shared +openmp +omptask +tests %gcc@10.3.1"
+    RUBY_BUILD_AND_TEST_JOB_ALLOC: "--time=60 --nodes=1"
+  extends: .build_and_test_on_ruby
+
+intel_19_1_2_gcc_8_5_0:
+  variables:
+    SPEC: " ~shared +openmp +omptask +tests %intel@19.1.2.gcc.8.5.0"
+    RUBY_BUILD_AND_TEST_JOB_ALLOC: "--time=90 --nodes=1"
+  extends: .build_and_test_on_ruby
```

Comment:
@rhornung67 is it OK with you that the old default is now expressed with "+omptask". I would have set it "on" by default.

Reply:
The truth is that it's not clear which option is better, but issues can arise when omptask is on. I switched the default to  avoid user breakages if possible. Now that it's exposed, users can choose.

Reply:
I understand. Here is my reasoning:

With your changes, we may want to turn omptask on for most of the CI specs if we do things like they used to be.
However, it looks like you restricted the variant to a subset of specs.

I think that ideally, spack default would be +omptask except with clang compilers.
With clango, ~omptask would be the default and +omptask would conflict.
This way, there is not preliminary knowledge required to build RAJA with any compiler, and the issue is exposed directly in the Spack package with a comment preceding the special case.

Reply:
@rhornung67 would you consider adding +omptask to the PROJECT_RUBY_VARIANTS (in .gitlab/custom-jobs-and-variables.yml). This would add it to all the shared spec, but things used to be that way.

Reply:
@adrienbernede let's leave it as is for now. I will be making another pass through the RADIUSS repos (adding new specs, etc. that are available) and RAJA and RAJAPerf CI where I will be adding more build and test variants.